app/src/main/java/ch/epfl/qedit/view/edit/EditQuizSettingsDialog.java

```diff
@@ -0,0 +1,294 @@
+     * @param stringPool The StringPool of the Quiz before the edition
+     * @param quiz The quiz at its state before the edition
+     * @return A new instance of such a dialog
+     */
+    public static EditQuizSettingsDialog newInstance(
+            SubmissionListener listener, StringPool stringPool, Quiz quiz) {
+        EditQuizSettingsDialog dialog = new EditQuizSettingsDialog();
+
+        Bundle args = new Bundle();
+        args.putSerializable(LISTENER_KEY, listener);
+
+        if (stringPool != null) {
+            args.putSerializable(STRING_POOL, stringPool);
+        }
+
+        if (quiz != null) {
+            args.putSerializable(QUIZ_ID, quiz);
+        }
+
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        // Setup basic elements of the dialog
+        View view = getArgumentsAndSetView();
+
+        if (quiz != null && stringPool != null) {
+            setupModifyingExistingQuiz(view);
+        } else {
+            // Edit a new quiz, so we need a new StringPool
+            stringPool = new StringPool();
+
+            setupLanguageSpinner(view);
+
+            // Setup treasure hunt
+            createTreasureHuntCheckbox(view);
+        }
+
+        // Set the title of the dialog
+        Spanned title =
+                Html.fromHtml(
+                        "<font color='#FF0000'>"
+                                + getString(R.string.edit_dialog_title_settings)
+                                + "</font>");
+
+        // Create a builder and build the dialog
+        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());
+        builder.setView(view)
+                .setTitle(title)
+                .setCancelable(true)
+                .setPositiveButton(
+                        R.string.button_start_editing_message,
+                        (dialog, id) -> prepareEditingAndSubmit())
+                .setNegativeButton(R.string.cancel, (dialog, which) -> dialog.cancel());
+        AlertDialog dialog = builder.create();
+
+        // Fix the behavior on changing title text
+        setupEditTitleTextWatcher(dialog);
+
+        return dialog;
+    }
+
+    public void setTextFilter(TextFilter textFilter) {
+        this.textFilter = Objects.requireNonNull(textFilter);
+    }
+
+    @Override
+    /* This method runs if the user selects another language */
+    public void onItemSelected(AdapterView parent, View view, int pos, long id) {
+        // Get language code from the position of the clicked language in the spinner
+        languageCode = getResources().getStringArray(R.array.languages_codes)[pos];
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView parent) {
+        // Not used because there will always be something selected
+    }
+
+    /**
+     * Gets all the arguments that are passes to the dialog, inflates the view and sets up the
```

Comment:
There are a few typos in this comment (I don't even know why I am looking at typos, but anyway).

Reply:
I'll correct the typos, thanks for point them out!